[Firebase_core]: Android Firestore initialization fails on second launch with ConcurrentModificationException

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Which plugins are affected?

Core

### Which platforms are affected?

Android

### Description

When running the app in debug mode on Android, the first launch works correctly and both Firestore databases are initialized as expected.
However, when I hot restart or relaunch the app in debug mode (while the previous debug session is still active), a PlatformException occurs during Firebase initialization with the following error:

java.util.ConcurrentModificationException
...
Caused by: java.util.ConcurrentModificationException
	at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin.lambda$didReinitializeFirebaseCore$1


The issue seems to happen when the app tries to reinitialize multiple Firestore instances for the same FirebaseApp.

Here’s a simplified version of the initialization code:

Future<void> initialize() async {
  try {
    final app = Firebase.app();

    final mainFirestore = FirebaseFirestore.instanceFor(
      app: app,
      databaseId: AppService.shared.config.databaseId,
    );

    mainFirestore.settings = const Settings(
      persistenceEnabled: true,
      cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED,
    );

    final appConfig = FirebaseFirestore.instanceFor(
      app: app,
      databaseId: "app-config",
    );

    appConfig.settings = const Settings(
      persistenceEnabled: true,
      cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED,
    );

    firestore = mainFirestore;
    appConfigFirestore = appConfig;

    logger.i('[FirestoreService] Initialization completed');
  } catch (e) {
    logger.e('[FirestoreService] Error during initialization: $e');
    rethrow;
  }
}


This method is called after:

await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);


Expected behavior:
Both Firestore instances (mainFirestore and appConfigFirestore) should initialize successfully on every app launch.

Actual behavior:
On the second launch in debug (or after a hot restart), the app crashes with a ConcurrentModificationException originating from Firebase Core or Firestore plugin internals.
It seems that the Firestore plugin tries to reinitialize its internal registry while multiple instances are being set up concurrently.

### Reproducing the issue

import 'package:flutter/material.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'firebase_options.dart';

class FirestoreService {
  late FirebaseFirestore firestore;
  late FirebaseFirestore appConfigFirestore;

  Future<void> initialize() async {
    try {
      final app = Firebase.app();

      final mainFirestore = FirebaseFirestore.instanceFor(
        app: app,
        databaseId: 'main-db',
      );

      mainFirestore.settings = const Settings(
        persistenceEnabled: true,
        cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED,
      );

      final appConfig = FirebaseFirestore.instanceFor(
        app: app,
        databaseId: 'app-config',
      );

      appConfig.settings = const Settings(
        persistenceEnabled: true,
        cacheSizeBytes: Settings.CACHE_SIZE_UNLIMITED,
      );

      firestore = mainFirestore;
      appConfigFirestore = appConfig;

      print('[FirestoreService] Initialization completed');
    } catch (e) {
      print('[FirestoreService] Error during initialization: $e');
      rethrow;
    }
  }
}

Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(options: DefaultFirebaseOptions.currentPlatform);
  await FirestoreService().initialize();
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Scaffold(
        body: Center(
          child: Text('Firestore multi-instance test'),
        ),
      ),
    );
  }
}


### Firebase Core version

4.2.0

### Flutter Version

3.32.8

### Relevant Log Output

```shell
>  [ERROR:flutter/runtime/dart_vm_initializer.cc(40)] Unhandled Exception: PlatformException(java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException, ExecutionException, Cause: java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException, Stacktrace: java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException
> E/flutter (18312): 	at com.google.android.gms.tasks.Tasks.zza(com.google.android.gms:play-services-tasks@@18.3.2:5)
> E/flutter (18312): 	at com.google.android.gms.tasks.Tasks.await(com.google.android.gms:play-services-tasks@@18.3.2:9)
> E/flutter (18312): 	at io.flutter.plugins.firebase.core.FlutterFirebaseCorePlugin.lambda$initializeCore$4(FlutterFirebaseCorePlugin.java:190)
> E/flutter (18312): 	at io.flutter.plugins.firebase.core.FlutterFirebaseCorePlugin.$r8$lambda$NxDsUkqe93KkHUjQxhXhbVz2y_4(Unknown Source:0)
> E/flutter (18312): 	at io.flutter.plugins.firebase.core.FlutterFirebaseCorePlugin$$ExternalSyntheticLambda7.run(D8$$SyntheticClass:0)
> E/flutter (18312): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1156)
> E/flutter (18312): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:651)
> E/flutter (18312): 	at java.lang.Thread.run(Thread.java:1119)
> E/flutter (18312): Caused by: java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException
> E/flutter (18312): 	at com.google.android.gms.tasks.Tasks.zza(com.google.android.gms:play-services-tasks@@18.3.2:5)
> E/flutter (18312): 	at com.google.android.gms.tasks.Tasks.await(com.google.android.gms:play-services-tasks@@18.3.2:9)
> E/flutter (18312): 	at io.flutter.plugins.firebase.core.FlutterFirebasePluginRegistry.lambda$didReinitializeFirebaseCore$1(FlutterFirebasePluginRegistry.java:82)
> E/flutter (18312): 	at io.flutter.plugins.firebase.core.FlutterFirebasePluginRegistry$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
> E/flutter (18312): 	... 3 more
> E/flutter (18312): Caused by: java.util.ConcurrentModificationException
> E/flutter (18312): 	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1603)
> E/flutter (18312): 	at java.util.HashMap$EntryIterator.next(HashMap.java:1636)
> E/flutter (18312): 	at java.util.HashMap$EntryIterator.next(HashMap.java:1634)
> E/flutter (18312): 	at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin.lambda$didReinitializeFirebaseCore$1(FlutterFirebaseFirestorePlugin.java:207)
> E/flutter (18312): 	at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin.$r8$lambda$LWPUzGoM67fJ0VnXoPogtvBOysE(Unknown Source:0)
> E/flutter (18312): 	at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestorePlugin$$ExternalSyntheticLambda5.run(D8$$SyntheticClass:0)
> E/flutter (18312): 	... 3 more
> E/flutter (18312): , null)
> E/flutter (18312): #0      FirebaseCoreHostApi.initializeCore (package:firebase_core_platform_interface/src/pigeon/messages.pigeon.dart:303:7)
> E/flutter (18312): <asynchronous suspension>
> E/flutter (18312): #1      MethodChannelFirebase._initializeCore (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:29:41)
> E/flutter (18312): <asynchronous suspension>
> E/flutter (18312): #2      MethodChannelFirebase.initializeApp (package:firebase_core_platform_interface/src/method_channel/method_channel_firebase.dart:70:7)
> E/flutter (18312): <asynchronous suspension>
> E/flutter (18312): #3      Firebase.initializeApp (package:firebase_core/src/firebase.dart:78:31)
> E/flutter (18312): <asynchronous suspension>
> E/flutter (18312): #4      runMainApp (package:myterritory/main.dart:20:3)
> E/flutter (18312): <asynchronous suspension>
> E/flutter (18312):
```

### Flutter dependencies

<!--- Look below for instructions on how to share your Flutter Dependencies. --->

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml
Dart SDK 3.8.1
Flutter SDK 3.32.8

dependencies:
- alert_info 0.0.6 [flutter]
- amazing_icons 3.0.1 [flutter jovial_svg]
- animated_segmented_tab_control 2.0.0 [flutter]
- app_badge_plus 1.2.3 [flutter plugin_platform_interface]
- app_links 6.4.1 [flutter app_links_linux app_links_platform_interface app_links_web]
- app_tracking_transparency 2.0.6+1 [flutter]
- build_runner 2.7.1 [args async build build_config build_daemon build_runner_core built_collection code_builder crypto dart_style frontend_server_client glob graphs http_multi_server io logging meta mime path pub_semver shelf shelf_web_socket stack_trace stream_transform watcher web_socket_channel yaml]
- carousel_slider 5.1.1 [flutter]
- cloud_firestore 6.0.3 [cloud_firestore_platform_interface cloud_firestore_web collection firebase_core firebase_core_platform_interface flutter meta]
- collection 1.19.1
- connectivity_plus 7.0.0 [flutter flutter_web_plugins connectivity_plus_platform_interface web meta nm collection]
- csv 6.0.0
- device_info_plus 12.2.0 [device_info_plus_platform_interface ffi file flutter flutter_web_plugins meta web win32 win32_registry]
- dio 5.9.0 [async collection http_parser meta mime path dio_web_adapter]
- dismiss_keyboard_on_tap 2.0.1 [flutter]
- extended_image 10.0.1 [extended_image_library flutter meta vector_math]
- file_picker 10.3.3 [flutter flutter_web_plugins flutter_plugin_android_lifecycle plugin_platform_interface ffi path win32 cross_file web dbus]
- firebase_analytics 12.0.3 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 6.1.1 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 4.2.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_crashlytics 5.0.3 [firebase_core firebase_core_platform_interface firebase_crashlytics_platform_interface flutter stack_trace]
- firebase_messaging 16.0.3 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_storage 13.0.3 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- fl_chart 1.0.0 [equatable flutter vector_math]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_animate 4.5.2 [flutter flutter_shaders]
- flutter_cached_pdfview 0.4.3 [flutter flutter_pdfview flutter_cache_manager path_provider path]
- flutter_datetime_picker_plus 2.2.0 [flutter]
- flutter_google_places_sdk 0.4.2+1 [flutter flutter_google_places_sdk_platform_interface flutter_google_places_sdk_ios flutter_google_places_sdk_web flutter_google_places_sdk_android flutter_google_places_sdk_windows flutter_google_places_sdk_linux flutter_google_places_sdk_macos]
- flutter_image_compress 2.4.0 [flutter flutter_image_compress_platform_interface flutter_image_compress_common flutter_image_compress_web flutter_image_compress_macos flutter_image_compress_ohos]
- flutter_inappwebview 6.1.5 [flutter flutter_inappwebview_platform_interface flutter_inappwebview_android flutter_inappwebview_ios flutter_inappwebview_macos flutter_inappwebview_web flutter_inappwebview_windows]
- flutter_keyboard_visibility 6.0.0 [meta flutter_keyboard_visibility_platform_interface flutter_keyboard_visibility_linux flutter_keyboard_visibility_macos flutter_keyboard_visibility_web flutter_keyboard_visibility_windows flutter]
- flutter_local_notifications 19.5.0 [clock flutter flutter_local_notifications_linux flutter_local_notifications_windows flutter_local_notifications_platform_interface timezone]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_pdfview 1.4.3 [flutter]
- flutter_polyline_points 3.1.0 [flutter http]
- flutter_svg 2.2.1 [flutter http vector_graphics vector_graphics_codec vector_graphics_compiler]
- flutter_switch 0.3.2 [flutter]
- geocoding 4.0.0 [flutter geocoding_platform_interface geocoding_android geocoding_ios]
- geolocator 14.0.2 [flutter geolocator_platform_interface geolocator_android geolocator_apple geolocator_web geolocator_windows geolocator_linux]
- get 5.0.0-release-candidate-9.3.2 [flutter flutter_web_plugins web]
- google_maps_cluster_manager_2 3.2.0 [flutter google_maps_flutter_platform_interface]
- google_maps_flutter 2.13.1 [flutter google_maps_flutter_android google_maps_flutter_ios google_maps_flutter_platform_interface google_maps_flutter_web]
- google_sign_in 7.2.0 [flutter google_sign_in_android google_sign_in_ios google_sign_in_platform_interface google_sign_in_web]
- grouped_list 6.0.0 [flutter]
- http 1.5.0 [async http_parser meta web]
- hugeicons 1.1.1 [flutter flutter_svg]
- in_app_review 2.0.11 [flutter in_app_review_platform_interface]
- intl 0.20.2 [clock meta path]
- ionicons 0.2.2 [flutter]
- isar_community 3.3.0-dev.3 [ffi js meta]
- isar_community_flutter_libs 3.3.0-dev.3 [flutter isar_community]
- location 8.0.1 [flutter location_platform_interface location_web]
- logger 2.6.2 [meta]
- lottie 3.3.1 [archive flutter http path vector_math]
- lucide_icons_flutter 3.1.5 [flutter]
- map_launcher 4.4.2 [flutter plugin_platform_interface]
- mobile_scanner 7.1.3 [collection flutter flutter_web_plugins meta plugin_platform_interface web]
- package_info_plus 8.3.1 [ffi flutter flutter_web_plugins http meta path package_info_plus_platform_interface web win32 clock]                                                                                       
- path_provider 2.1.5 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]                                                             
- pdf 3.11.3 [archive barcode bidi crypto image meta path_parsing vector_math xml]
- permission_handler 12.0.1 [flutter meta permission_handler_android permission_handler_apple permission_handler_html permission_handler_windows permission_handler_platform_interface]                               
- qr_flutter 4.1.0 [flutter qr]
- screen_brightness 2.1.7 [flutter screen_brightness_platform_interface screen_brightness_android screen_brightness_ios screen_brightness_macos screen_brightness_windows screen_brightness_ohos]                     
- share_plus 12.0.1 [cross_file meta mime flutter flutter_web_plugins share_plus_platform_interface file url_launcher_web url_launcher_windows url_launcher_linux url_launcher_platform_interface ffi web win32]      
- shared_preferences 2.5.3 [flutter shared_preferences_android shared_preferences_foundation shared_preferences_linux shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]        
- sign_in_with_apple 7.0.1 [flutter meta sign_in_with_apple_platform_interface sign_in_with_apple_web]
- smooth_page_indicator 1.2.1 [flutter]
- smooth_sheets2 0.10.0 [collection flutter meta]
- top_snackbar_flutter 3.3.0 [flutter]
- tutorial_coach_mark 1.3.3 [flutter]
- url_launcher 6.3.2 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]                                      
- uuid 4.5.1 [crypto sprintf meta fixnum]
- video_player 2.10.0 [flutter html video_player_android video_player_avfoundation video_player_platform_interface video_player_web]                                                                                  
- xml 6.6.1 [collection meta petitparser]

dev dependencies:
- flutter_lints 6.0.0 [lints]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math leak_tracker_flutter_testing async boolean_selector characters collection leak_tracker leak_tracker_testing material_color_utilities meta source_span stream_channel string_scanner term_glyph vm_service]                          
- isar_community_generator 3.3.0-dev.3 [analyzer build dart_style dartx glob isar_community path source_gen xxh3]                                                                                                     

transitive dependencies:
- _fe_analyzer_shared 88.0.0 [meta]
- _flutterfire_internals 1.3.63 [collection firebase_core firebase_core_platform_interface flutter meta]
- analyzer 8.1.1 [_fe_analyzer_shared collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]                                                                                    
- app_links_linux 1.0.3 [flutter app_links_platform_interface gtk]
- app_links_platform_interface 2.0.2 [flutter plugin_platform_interface]
- app_links_web 1.0.4 [flutter flutter_web_plugins app_links_platform_interface web]
- archive 4.0.7 [crypto path posix]
- args 2.7.0
- async 2.13.0 [collection meta]
- barcode 2.2.9 [meta qr]
- bidi 2.0.13
- boolean_selector 2.1.2 [source_span string_scanner]
- build 3.1.0 [analyzer async build_runner_core convert crypto glob logging package_config path]
- build_config 1.2.0 [checked_yaml json_annotation path pubspec_parse]
- build_daemon 4.0.4 [built_collection built_value crypto http_multi_server logging path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]                                                     
- build_resolvers 3.0.3 [analyzer async build build_runner_core collection package_config path pool pub_semver]                                                                                                       
- build_runner_core 9.3.1 [analyzer async build build_config build_resolvers build_runner built_collection built_value collection convert crypto glob graphs json_annotation logging meta package_config path pool timing watcher yaml]                                                                                          
- built_collection 5.1.1
- built_value 8.12.0 [built_collection collection fixnum meta]
- characters 1.4.0
- checked_yaml 2.0.4 [json_annotation source_span yaml]
- clock 1.1.2
- cloud_firestore_platform_interface 7.0.3 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]                                                                                   
- cloud_firestore_web 5.0.3 [_flutterfire_internals cloud_firestore_platform_interface collection firebase_core firebase_core_web flutter flutter_web_plugins]                                                        
- code_builder 4.11.0 [built_collection built_value collection matcher meta]
- connectivity_plus_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- convert 3.1.2 [typed_data]
- cross_file 0.3.4+2 [meta web]
- crypto 3.0.6 [typed_data]
- csslib 1.0.2 [source_span]
- dart_style 3.1.2 [analyzer args collection package_config path pub_semver source_span yaml]
- dartx 1.2.0 [characters collection crypto meta path time]
- dbus 0.7.11 [args ffi meta xml]
- device_info_plus_platform_interface 7.0.3 [flutter meta plugin_platform_interface]
- dio_web_adapter 2.1.1 [dio http_parser meta web]
- equatable 2.0.7 [collection meta]
- extended_image_library 5.0.1 [crypto flutter http_client_helper js path path_provider web]
- fake_async 1.3.3 [clock collection]
- ffi 2.1.4
- file 7.0.1 [meta path]
- firebase_analytics_platform_interface 5.0.3 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]                                                                                           
- firebase_analytics_web 0.6.0+3 [_flutterfire_internals firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins]                                                           
- firebase_auth_platform_interface 8.1.3 [_flutterfire_internals collection firebase_core flutter http meta plugin_platform_interface]                                                                                
- firebase_auth_web 6.0.4 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser meta web]                                                                         
- firebase_core_platform_interface 6.0.2 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 3.2.0 [firebase_core_platform_interface flutter flutter_web_plugins meta web]
- firebase_crashlytics_platform_interface 3.8.14 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]                                                                             
- firebase_messaging_platform_interface 4.7.3 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]                                                                                           
- firebase_messaging_web 4.0.3 [_flutterfire_internals firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins meta web]                                                    
- firebase_storage_platform_interface 5.2.14 [_flutterfire_internals collection firebase_core flutter meta plugin_platform_interface]                                                                                 
- firebase_storage_web 3.10.21 [_flutterfire_internals async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http meta web]                                           
- fixnum 1.1.1
- flutter_cache_manager 3.4.1 [clock collection file flutter http path path_provider rxdart sqflite uuid]
- flutter_google_places_sdk_android 0.2.0 [flutter flutter_google_places_sdk_platform_interface]
- flutter_google_places_sdk_http 0.2.1 [flutter http latlong2 flutter_google_places_sdk_platform_interface freezed_annotation json_annotation]                                                                        
- flutter_google_places_sdk_ios 0.2.0 [flutter flutter_google_places_sdk_platform_interface]
- flutter_google_places_sdk_linux 0.2.1 [flutter flutter_google_places_sdk_platform_interface flutter_google_places_sdk_http]                                                                                         
- flutter_google_places_sdk_macos 0.2.1 [flutter flutter_google_places_sdk_platform_interface flutter_google_places_sdk_http]                                                                                         
- flutter_google_places_sdk_platform_interface 0.3.2+1 [flutter freezed_annotation json_annotation plugin_platform_interface]                                                                                         
- flutter_google_places_sdk_web 0.2.0 [flutter flutter_web_plugins flutter_google_places_sdk_platform_interface google_maps collection web]                                                                           
- flutter_google_places_sdk_windows 0.2.1 [flutter flutter_google_places_sdk_platform_interface flutter_google_places_sdk_http]                                                                                       
- flutter_image_compress_common 1.0.6 [flutter flutter_image_compress_platform_interface]
- flutter_image_compress_macos 1.0.3 [flutter flutter_image_compress_platform_interface]
- flutter_image_compress_ohos 0.0.3 [flutter flutter_image_compress_platform_interface]
- flutter_image_compress_platform_interface 1.0.5 [flutter plugin_platform_interface cross_file]
- flutter_image_compress_web 0.1.5 [flutter flutter_web_plugins flutter_image_compress_platform_interface web]                                                                                                        
- flutter_inappwebview_android 1.1.3 [flutter flutter_inappwebview_platform_interface]
- flutter_inappwebview_internal_annotations 1.2.0
- flutter_inappwebview_ios 1.1.2 [flutter flutter_inappwebview_platform_interface]
- flutter_inappwebview_macos 1.1.2 [flutter flutter_inappwebview_platform_interface]
- flutter_inappwebview_platform_interface 1.3.0+1 [flutter flutter_inappwebview_internal_annotations plugin_platform_interface]                                                                                       
- flutter_inappwebview_web 1.1.2 [flutter flutter_web_plugins web flutter_inappwebview_platform_interface]
- flutter_inappwebview_windows 0.6.0 [flutter flutter_inappwebview_platform_interface]
- flutter_keyboard_visibility_linux 1.0.0 [flutter_keyboard_visibility_platform_interface flutter]
- flutter_keyboard_visibility_macos 1.0.0 [flutter_keyboard_visibility_platform_interface flutter]
- flutter_keyboard_visibility_platform_interface 2.0.0 [flutter meta plugin_platform_interface]
- flutter_keyboard_visibility_web 2.0.0 [flutter_keyboard_visibility_platform_interface flutter_web_plugins flutter]                                                                                                  
- flutter_keyboard_visibility_windows 1.0.0 [flutter_keyboard_visibility_platform_interface flutter]
- flutter_local_notifications_linux 6.0.0 [dbus ffi flutter flutter_local_notifications_platform_interface path xdg_directories]                                                                                      
- flutter_local_notifications_platform_interface 9.1.0 [plugin_platform_interface]
- flutter_local_notifications_windows 1.0.3 [flutter ffi flutter_local_notifications_platform_interface meta timezone xml]                                                                                            
- flutter_plugin_android_lifecycle 2.0.30 [flutter]
- flutter_shaders 0.1.3 [flutter vector_math]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math]
- freezed_annotation 3.1.0 [collection json_annotation meta]
- frontend_server_client 4.0.0 [async path]
- geoclue 0.1.1 [dbus meta]
- geocoding_android 4.0.1 [flutter geocoding_platform_interface]
- geocoding_ios 3.1.0 [flutter geocoding_platform_interface]
- geocoding_platform_interface 3.2.0 [flutter meta plugin_platform_interface]
- geolocator_android 5.0.2 [flutter geolocator_platform_interface meta uuid]
- geolocator_apple 2.3.13 [flutter geolocator_platform_interface]
- geolocator_linux 0.2.3 [dbus flutter geoclue geolocator_platform_interface gsettings package_info_plus]
- geolocator_platform_interface 4.2.6 [flutter plugin_platform_interface vector_math meta]
- geolocator_web 4.1.3 [flutter flutter_web_plugins geolocator_platform_interface web]
- geolocator_windows 0.2.5 [flutter geolocator_platform_interface]
- glob 2.1.3 [async collection file path string_scanner]
- google_identity_services_web 0.3.3+1 [meta web]
- google_maps 8.2.0 [meta web]
- google_maps_flutter_android 2.18.2 [flutter flutter_plugin_android_lifecycle google_maps_flutter_platform_interface stream_transform]                                                                               
- google_maps_flutter_ios 2.15.5 [flutter google_maps_flutter_platform_interface stream_transform]
- google_maps_flutter_platform_interface 2.14.0 [collection flutter plugin_platform_interface stream_transform]                                                                                                       
- google_maps_flutter_web 0.5.14+2 [collection flutter flutter_web_plugins google_maps google_maps_flutter_platform_interface sanitize_html stream_transform web]                                                     
- google_sign_in_android 7.2.0 [flutter google_sign_in_platform_interface]
- google_sign_in_ios 6.2.1 [flutter google_sign_in_platform_interface]
- google_sign_in_platform_interface 3.1.0 [flutter plugin_platform_interface]
- google_sign_in_web 1.1.0 [flutter flutter_web_plugins google_identity_services_web google_sign_in_platform_interface http web]                                                                                      
- graphs 2.3.2 [collection]
- gsettings 0.2.8 [dbus xdg_directories]
- gtk 2.1.0 [ffi flutter meta]
- html 0.15.6 [csslib source_span]
- http_client_helper 3.0.0 [http]
- http_multi_server 3.2.2 [async]
- http_parser 4.1.2 [collection source_span string_scanner typed_data]
- image 4.5.4 [archive meta xml]
- in_app_review_platform_interface 2.0.5 [flutter url_launcher plugin_platform_interface platform]
- io 1.0.5 [meta path string_scanner]
- jovial_misc 0.9.2 [async collection convert meta]
- jovial_svg 1.1.28 [xml collection args vector_math jovial_misc http meta flutter]
- js 0.7.2
- json_annotation 4.9.0 [meta]
- latlong2 0.9.1 [intl]
- leak_tracker 10.0.9 [clock collection meta path vm_service]
- leak_tracker_flutter_testing 3.0.9 [flutter leak_tracker leak_tracker_testing matcher meta]
- leak_tracker_testing 3.0.1 [leak_tracker matcher meta]
- lints 6.0.0
- location_platform_interface 6.0.1 [flutter plugin_platform_interface]
- location_web 6.0.1 [flutter flutter_web_plugins http_parser location_platform_interface web]
- logging 1.3.0
- matcher 0.12.17 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.11.1 [collection]
- meta 1.16.0
- mime 2.0.0
- nm 0.5.0 [dbus]
- package_config 2.2.0 [path]
- package_info_plus_platform_interface 3.2.1 [flutter meta plugin_platform_interface]
- path 1.9.1
- path_parsing 1.1.0 [meta vector_math]
- path_provider_android 2.2.18 [flutter path_provider_platform_interface]
- path_provider_foundation 2.4.2 [flutter path_provider_platform_interface]
- path_provider_linux 2.2.1 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.1.2 [flutter platform plugin_platform_interface]
- path_provider_windows 2.3.0 [ffi flutter path path_provider_platform_interface]
- permission_handler_android 13.0.1 [flutter permission_handler_platform_interface]
- permission_handler_apple 9.4.7 [flutter permission_handler_platform_interface]
- permission_handler_html 0.1.3+5 [flutter flutter_web_plugins permission_handler_platform_interface web]
- permission_handler_platform_interface 4.3.0 [flutter meta plugin_platform_interface]
- permission_handler_windows 0.2.1 [flutter permission_handler_platform_interface]
- petitparser 7.0.1 [meta collection]
- platform 3.1.6
- plugin_platform_interface 2.1.8 [meta]
- pool 1.5.2 [async stack_trace]
- posix 6.0.3 [ffi meta path]
- pub_semver 2.2.0 [collection]
- pubspec_parse 1.5.0 [checked_yaml collection json_annotation pub_semver yaml]
- qr 3.0.2 [meta]
- rxdart 0.28.0
- sanitize_html 2.1.0 [html meta]
- screen_brightness_android 2.1.3 [flutter screen_brightness_platform_interface]
- screen_brightness_ios 2.1.2 [flutter screen_brightness_platform_interface]
- screen_brightness_macos 2.1.1 [flutter screen_brightness_platform_interface]
- screen_brightness_ohos 2.1.2 [flutter screen_brightness_platform_interface]
- screen_brightness_platform_interface 2.1.0 [flutter plugin_platform_interface]
- screen_brightness_windows 2.1.0 [flutter screen_brightness_platform_interface]
- share_plus_platform_interface 6.1.0 [cross_file flutter meta mime plugin_platform_interface path_provider uuid]                                                                                                     
- shared_preferences_android 2.4.13 [flutter shared_preferences_platform_interface]
- shared_preferences_foundation 2.5.4 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.4.1 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]                                                                       
- shared_preferences_platform_interface 2.4.1 [flutter plugin_platform_interface]
- shared_preferences_web 2.4.3 [flutter flutter_web_plugins shared_preferences_platform_interface web]
- shared_preferences_windows 2.4.1 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]                                                                   
- shelf 1.4.2 [async collection http_parser path stack_trace stream_channel]
- shelf_web_socket 3.0.0 [shelf stream_channel web_socket_channel]
- sign_in_with_apple_platform_interface 2.0.0 [flutter plugin_platform_interface meta]
- sign_in_with_apple_web 3.0.0 [flutter flutter_web_plugins sign_in_with_apple_platform_interface]
- sky_engine 0.0.0
- source_gen 3.1.0 [analyzer async build dart_style glob path pub_semver source_span yaml]
- source_span 1.10.1 [collection path term_glyph]
- sprintf 7.0.0
- sqflite 2.4.2 [flutter sqflite_android sqflite_darwin sqflite_platform_interface sqflite_common path]
- sqflite_android 2.4.1 [flutter sqflite_common path sqflite_platform_interface]
- sqflite_common 2.5.6 [synchronized path meta]
- sqflite_darwin 2.4.2 [flutter sqflite_platform_interface meta sqflite_common path]
- sqflite_platform_interface 2.4.0 [flutter platform sqflite_common plugin_platform_interface meta]
- stack_trace 1.12.1 [path]
- stream_channel 2.1.4 [async]
- stream_transform 2.1.1
- string_scanner 1.4.1 [source_span]
- synchronized 3.4.0
- term_glyph 1.2.2
- test_api 0.7.4 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]                                                                                            
- time 2.1.5 [clock]
- timezone 0.10.1 [http path]
- timing 1.0.2 [json_annotation]
- typed_data 1.4.0 [collection]
- url_launcher_android 6.3.20 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.3.4 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.2.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.2.3 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.3.2 [flutter plugin_platform_interface]
- url_launcher_web 2.4.1 [flutter flutter_web_plugins url_launcher_platform_interface web]
- url_launcher_windows 3.1.4 [flutter url_launcher_platform_interface]
- vector_graphics 1.1.19 [flutter http vector_graphics_codec]
- vector_graphics_codec 1.1.13
- vector_graphics_compiler 1.1.19 [args meta path path_parsing vector_graphics_codec xml]
- vector_math 2.1.4
- video_player_android 2.8.14 [flutter video_player_platform_interface]
- video_player_avfoundation 2.8.4 [flutter video_player_platform_interface]
- video_player_platform_interface 6.4.0 [flutter plugin_platform_interface]
- video_player_web 2.4.0 [flutter flutter_web_plugins video_player_platform_interface web]
- vm_service 15.0.0
- watcher 1.1.4 [async path]
- web 1.1.1
- web_socket 1.0.1 [web]
- web_socket_channel 3.0.3 [async crypto stream_channel web web_socket]
- win32 5.14.0 [ffi]
- win32_registry 2.1.0 [ffi meta win32]
- xdg_directories 1.1.0 [meta path]
- xxh3 1.2.0
- yaml 3.1.3 [collection source_span string_scanner]

```

</details>


### Additional context and comments

_No response_

--- a/packages/firebase_database/firebase_database/android/src/main/kotlin/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.kt
+++ b/packages/firebase_database/firebase_database/android/src/main/kotlin/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.kt
@@ -959,7 +959,7 @@ class FirebaseDatabasePlugin :
       val streamHandler = EventStreamHandler(query, object : OnDispose {
         override fun run() {
           // Clean up when the stream is disposed
-          streamHandlers.remove(eventChannel)
+         eventChannel.setStreamHandler(null)
         }
       })
       eventChannel.setStreamHandler(streamHandler)

